### PR TITLE
[TAF] Added question scenarios

### PR DESCRIPTION
**Pull request by sydorenkooleksii**

### CODE REVIEW

File: src/main/java/automation/examples/site/components/question/AddedQuestionFragment.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package automation.examples.site.components.question;
+
+import automation.examples.site.components.Fragment;
+import automation.examples.storage.models.Question;
+
+public interface AddedQuestionFragment extends Fragment {
+
+    Question getAsAddQuestion();
```
Comment (aqaexplorer) [inline src/main/java/automation/examples/site/components/question/AddedQuestionFragment.java]: suggestion to change on 'getAsQuestion'

**Review by YuriiKononenko — APPROVED** [review]

File: src/test/resources/features/Question.feature [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  Background:
+    Given logged in User
+    And User has added new competition with Javascript title, Mentoring description and Education category
+
+  @type:sunny @role=admin @WIP
+  Scenario: Add question
+    Given User navigated to first competition on Landing Page
+    And has provided 'what is Javascript' question with 'programming language' answear, 'base' type and 'medium' value
+    When User seves question
```
Comment (yuriytkachenko-ideals) [inline src/test/resources/features/Question.feature]: Check step. "seves"

**Review by yuriytkachenko-ideals — CHANGES_REQUESTED** [review]

**Review by yuriytkachenko-ideals — APPROVED** [review]

**Review by aqaexplorer — APPROVED** [review]

File: src/main/java/automation/examples/site/components/question/AddedQuestionFragment.java (hunk 1 above)
Comment (sydorenkooleksii) [reply src/main/java/automation/examples/site/components/question/AddedQuestionFragment.java]: Thank you. Done

File: src/test/resources/features/Question.feature (hunk 2 above)
Comment (sydorenkooleksii) [reply src/test/resources/features/Question.feature]: Thank you. Done

File: src/main/java/automation/examples/site/pages/home/HomePage.java [hunk 3]
```diff
@@ -1,12 +1,24 @@
...
+
+    SelenideElement getLogoLink();
+
+    void navigateViaLogoLink();
+
+    boolean isProfileLinkEnabled();
+
+    String getProfileEmail();
+
+    void clickCompetitionLink(final String name);
```
Comment (YuriiKononenko) [inline src/main/java/automation/examples/site/pages/home/HomePage.java]: check signature pls

File: src/test/java/automation/examples/storefront/business/TrainingPageDefinitionSteps.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    public void saveQuestion() {
+        trainingPage.getQuestionFragment().saveQuestion();
+    }
+
+    @When("^User cancel question creation$")
+    public void cancelQuestion() {
+        trainingPage.getQuestionFragment().cancelQuestion();
+    }
+
+    @When("^User cancels deletion previously added question$")
```
Comment (YuriiKononenko) [inline src/test/java/automation/examples/storefront/business/TrainingPageDefinitionSteps.java]: I would better add `deletion of`

**Review by YuriiKononenko — APPROVED** [review]

File: src/main/java/automation/examples/site/pages/home/HomePageImpl.java [hunk 5]
```diff
@@ -1,21 +1,36 @@
...
+    private LoginFragment loginFragment;
+
+    @Override
+    public LoginFragment getLoginFragment() {
+        return loginFragment;
+    }
+
     @Override
     public void openPage() {
         logger.debug("Logger works!");
```
Comment (workshy) [inline src/main/java/automation/examples/site/pages/home/HomePageImpl.java:36]: This is useless log message. Paraphrase pls.

File: src/test/java/automation/examples/storefront/RunCucumberTest.java [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
+package automation.examples;
+
+import cucumber.api.CucumberOptions;
+import cucumber.api.junit.Cucumber;
+import org.junit.runner.RunWith;
+
+@RunWith(Cucumber.class)
+@CucumberOptions(
+        glue = "automation.examples.storefront.business",
+        features = "src/test/resources/features/storefront",
```
Comment (workshy) [inline src/test/java/automation/examples/storefront/RunCucumberTest.java:10]: Add the same changes to profile in pom pls

File: src/main/java/automation/examples/site/pages/home/HomePage.java (hunk 3 above)
Comment (sydorenkooleksii) [reply src/main/java/automation/examples/site/pages/home/HomePage.java]: Thank you. Done

File: src/test/java/automation/examples/storefront/business/TrainingPageDefinitionSteps.java (hunk 4 above)
Comment (sydorenkooleksii) [reply src/test/java/automation/examples/storefront/business/TrainingPageDefinitionSteps.java]: Thank you. Done

File: src/main/java/automation/examples/site/pages/home/HomePageImpl.java (hunk 5 above)
Comment (sydorenkooleksii) [reply src/main/java/automation/examples/site/pages/home/HomePageImpl.java:36]: This logger will be used in future projects. And added like template by @yuz123

File: src/test/java/automation/examples/storefront/RunCucumberTest.java (hunk 6 above)
Comment (sydorenkooleksii) [reply src/test/java/automation/examples/storefront/RunCucumberTest.java:10]: @YuriiKononenko will add profiles in pom

**Review by workshy — APPROVED** [review]

**Pull request merged**